joern-cli/frontends/pysrc2cpg/src/main/scala/io/joern/pysrc2cpg/PythonTypeHintCallLinker.scala

```diff
@@ -18,4 +21,41 @@ class PythonTypeHintCallLinker(cpg: Cpg) extends XTypeHintCallLinker(cpg) {
     case typ => typ
   }
 
+  override def linkCalls(builder: DiffGraphBuilder): Unit = {
+    val methodMap = mutable.HashMap.empty[String, MethodBase]
+    val callerAndCallees = calls
+      .map(call => (call, calleeNames(call)))
+      .toList
+    // Gather all method nodes and/or stubs
+    callerAndCallees
+      .foreach { case (call, methodNames) =>
+        val ms = callees(methodNames).l
+        if (ms.nonEmpty) {
+          ms.foreach { m => methodMap.put(m.fullName, m) }
+        } else {
+          val mNames = ms.map(_.fullName).toSet
+          methodNames
+            .filterNot(mNames.contains)
+            .map(fullName => createMethodStub(fullName, call, builder))
+            .foreach { m => methodMap.put(m.fullName, m) }
+        }
+      }
+    // Link edges to method nodes
+    callerAndCallees.foreach { case (call, methodNames) =>
+      methodNames
+        .flatMap(methodMap.get)
+        .foreach { m => builder.addEdge(call, m, EdgeTypes.CALL) }
+      if (methodNames.size == 1) {
+        builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, methodNames.head)
+      } else if (methodNames.size > 1) {
+        val builtInMethodName = methodNames.filter(s => s.contains("__builtin"))
+        if (builtInMethodName.size > 0) {
+          builder.setNodeProperty(call, PropertyNames.METHOD_FULL_NAME, builtInMethodName.sortBy(_.length).head)
```

Comment:
Perhaps one additional heuristic we could use, is to see if there is a full name that does not contain dummy values. I.e.
```object XTypeRecovery {
  val DummyReturnType  = "<returnValue>"
  val DummyMemberLoad  = "<member>"
  val DummyIndexAccess = "<indexAccess>"
  ...
```
Perhaps you could check if the string contains any of these and filter these out. If there is a full path that exists after the filter, then choose that.